### PR TITLE
add dante support to haskell layer

### PR DESCRIPTION
**Pull request by mckeankylej**
This PR adds [dante](https://github.com/jyp/dante) support to the haskell layer.
Dante has all the features of intero without the dep of stack! It works with plain cabal, nix, styx, and stack projects.

### CODE REVIEW

**Comment by mckeankylej** [comment]
Any reason this isn't merged yet? I want to let my coworkers go off of my own spacemacs branch.

File: layers/+lang/haskell/funcs.el [hunk 1]
```diff
@@ -81,6 +96,12 @@
   (evil-define-key '(insert normal) intero-mode-map
     (kbd "M-.") 'intero-goto-definition))
 
+;; Dante Functions
+
+(defun haskell-dante/insert-type ()
```
Comment (TheBB) [inline layers/+lang/haskell/funcs.el]: Please prefix with `spacemacs-haskell`, like the other functions.

File: layers/+lang/haskell/packages.el [hunk 2]
```diff
@@ -54,6 +55,13 @@
   (use-package ghc
     :defer t))
 
+(defun haskell/init-dante ()
+  (use-package dante
+    :defer t
+    :commands 'dante-mode)
+  :init
```
Comment (TheBB) [inline layers/+lang/haskell/packages.el]: This `:init` block has to be in the `use-package` call.

File: layers/+lang/haskell/packages.el [hunk 3]
```diff
@@ -54,6 +55,13 @@
   (use-package ghc
     :defer t))
 
+(defun haskell/init-dante ()
+  (use-package dante
+    :defer t
+    :commands 'dante-mode)
```
Comment (TheBB) [inline layers/+lang/haskell/packages.el]: `dante-mode` should not be quoted, but actually this line is not necessary, as `dante-mode` [is autoloaded](https://github.com/jyp/dante/blob/master/dante.el#L157-L158).

**Comment by mckeankylej** [comment]
I added your request changes and also rebased onto spacemacs/develop

**Comment by JAremko** [comment]
@mckeankylej  update TOC https://travis-ci.org/syl20bnr/spacemacs/jobs/235660780#L479

**Comment by mckeankylej** [comment]
What is TOC @JAremko

sorry this is my first PR to spacemacs

**Comment by JAremko** [comment]
@mckeankylej  [Table of Contents]( https://github.com/mckeankylej/spacemacs/blob/b70a3aec48dc44bc7a0a274238c5f70ab64366f1/layers/%2Blang/haskell/README.org#table-of-contents) You need to add [`dante`](https://github.com/mckeankylej/spacemacs/blob/b70a3aec48dc44bc7a0a274238c5f70ab64366f1/layers/%2Blang/haskell/README.org#dante) there. 

See  https://travis-ci.org/syl20bnr/spacemacs/jobs/235660780#L479

**Comment by mckeankylej** [comment]
Those new travis failures are not in the haskell lang directory should I fix them?

**Comment by JAremko** [comment]
@mckeankylej  What the... :cold_sweat: Looks like someone broke something.. Hmmm.  

You shouldn't fix it but you can  rebase your PR

**Comment by mckeankylej** [comment]
@JAremko rebase it onto what? The latest develop?

**Comment by JAremko** [comment]
@mckeankylej  yeah.  And squash it all into one-two commits. I think one is better :smile:

**Comment by mckeankylej** [comment]
@JAremko thanks for all the help could you give me some resources on squashing commits? I have never done this before.

**Comment by JAremko** [comment]
@mckeankylej 

`git rebase  -i HEAD~5`

you'll see last 5 commits.  Replace `pick` with `squash`  or `s`  before  commits that you want to squash. 
**DON'T squash the first commit**
then just save annnd `git push --force`

**Comment by mckeankylej** [comment]
@JAremko yea I tried that early the problem is that goes back to commits that I didnt pr. I shouldnt be rebasing commits I didnt commit right?

**Comment by mckeankylej** [comment]
lol there goes all my work

**Comment by mckeankylej** [comment]
oh wait no that worked

**Comment by JAremko** [comment]
what ` layers/+lang/semantic-web/README.org` doing in there :cold_sweat:

**Comment by mckeankylej** [comment]
@JAremko should I just re clone spacemacs develop?

**Comment by JAremko** [comment]
@mckeankylej  you can copy all files that you want to change into some safe place.  Then `git reset --hard upstream/develop` and copy them back and commit.

**Comment by JAremko** [comment]
Something like this should do:
```
git remote add upstream https://github.com/syl20bnr/spacemacs.git
git fetch upstream/develop
git reset --hard upstream/develop
<add files>
git add --all
git commit -m "add dante support to haskell layer"
git push --force
```

**Comment by JAremko** [comment]
YAY! You did it! :clap:

**Comment by mckeankylej** [comment]
Thanks, ❤️ spacemacs and spacemacs devs! Tests passed too!

**Comment by JAremko** [comment]
@TheBB  I think we are ready to go :rocket:

File: layers/+lang/haskell/packages.el [hunk 4]
```diff
@@ -54,6 +55,11 @@
   (use-package ghc
     :defer t))
 
+(defun haskell/init-dante ()
+  (use-package dante
+    :defer t
+    :init (add-hook 'haskell-mode-hook 'dante-mode)))
```
Comment (TheBB) [inline layers/+lang/haskell/packages.el]: Final question: Shouldn't enabling `dante-mode` in a specific haskell buffer be the responsibility of the `setup-dante` function?

**Comment by TheBB** [comment]
Yeah, looks good for the most part. Just wondering about that `add-hook`.

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

I went ahead and moved `dante-mode` as I said. I checked the source and all it does is change the flycheck checker. Seems ok.

**Pull request closed without merging**